modules/flocIdSystem.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * This module adds flocId to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/flocId
+ * @requires module:modules/userId
+ */
+
+import * as utils from '../src/utils.js'
+import {submodule} from '../src/hook.js'
+
+const MODULE_NAME = 'originFloc';
+
+/**
+ * Add meta tag to support enabling of floc origin trial
+ * @function
+ * @param {string} token - configured token for origin-trial
+ */
+function enableOriginTrial(token) {
+  const tokenElement = document.createElement('meta')
+  tokenElement.httpEquiv = 'origin-trial'
+  tokenElement.content = token
+  document.head.appendChild(tokenElement)
+}
+
+/**
+ * Encode the id
+ * @param value
+ * @returns {string|*}
+ */
+function encodeId(value) {
+  const result = {};
+  if (value) {
+    result.flocId = value;
+    utils.logInfo('Decoded value ' + JSON.stringify(result));
+    return result;
+  }
+  return undefined;
+}
+
+/** @type {Submodule} */
+export const flocIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param {string} value
+   * @returns {{flocId:{ id: string }} or undefined if value doesn't exists
+   */
+  decode(value) {
+    return (value) ? encodeId(value) : undefined;
+  },
+  /**
+   * If chrome and cohort enabled performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleConfig} [config]
+   * @returns {IdResponse|undefined}
+   */
+  getId(config) {
+    const isChrome = !!window.chrome && (!!window.chrome.webstore || !!window.chrome.runtime);
+
+    // Validate feature is enabled
+    const isFlocEnabled = !!document.featurePolicy && !!document.featurePolicy.features() && document.featurePolicy.features().includes('interest-cohort');
+
+    if (isChrome && isFlocEnabled) {
+      const configParams = (config && config.params) || {};
+      if (!configParams || (typeof configParams.token !== 'string')) {
+        utils.logInfo('User ID - flocId submodule token is not defined');
+      }
+
+      // Insert meta-tag with token from configuration
+      enableOriginTrial(configParams.token);
```

Comment:
this module should work with no token, as the token will no longer be required after July. It seems like this function should be behind the conditional for is the token defined.